src/main/java/com/yelp/nrtsearch/tools/nrt_utils/state/StateCommandUtils.java

```diff
@@ -0,0 +1,370 @@
+          AmazonS3ClientBuilder.standard().withCredentials(awsCredentialsProvider).build();
+      clientRegion = s3ClientInterim.getBucketLocation(bucketName);
+      // In useast-1, the region is returned as "US" which is an equivalent to "us-east-1"
+      // https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/Region.html#US_Standard
+      // However, this causes an UnknownHostException so we override it to the full region name
+      if (clientRegion.equals("US")) {
+        clientRegion = "us-east-1";
+      }
+    } else {
+      clientRegion = region;
+    }
+    String serviceEndpoint = String.format("s3.%s.amazonaws.com", clientRegion);
+    System.out.printf("S3 ServiceEndpoint: %s%n", serviceEndpoint);
+    return AmazonS3ClientBuilder.standard()
+        .withCredentials(awsCredentialsProvider)
+        .withEndpointConfiguration(
+            new AwsClientBuilder.EndpointConfiguration(serviceEndpoint, clientRegion))
+        .build();
+  }
+
+  /**
+   * Get the contents of the state file stored in remote state.
+   *
+   * @param versionManager state version manager
+   * @param serviceName nrtsearch cluster service name
+   * @param resourceName state resource name, for index this must include unique identifier
+   * @param stateFileName name of state file in archive
+   * @return state file contents as string
+   * @throws IOException
+   */
+  public static String getStateFileContents(
+      VersionManager versionManager, String serviceName, String resourceName, String stateFileName)
+      throws IOException {
+    String backendResourceName =
+        IndexBackupUtils.isBackendGlobalState(resourceName)
+            ? resourceName
+            : getIndexStateResource(resourceName);
+    long currentVersion = versionManager.getLatestVersionNumber(serviceName, backendResourceName);
+    if (currentVersion == -1) {
+      System.out.println(
+          "No state exists for service: " + serviceName + ", resource: " + backendResourceName);
+      return null;
+    }
+    System.out.println("Current version: " + currentVersion);
+    String versionStr =
+        versionManager.getVersionString(
+            serviceName, backendResourceName, String.valueOf(currentVersion));
+    System.out.println("Version string: " + versionStr);
+
+    String absoluteResourcePath = getStateKey(serviceName, backendResourceName, versionStr);
+    if (!versionManager
+        .getS3()
+        .doesObjectExist(versionManager.getBucketName(), absoluteResourcePath)) {
+      System.out.println("Resource does not exist: " + absoluteResourcePath);
+      return null;
+    }
+
+    String stateStr =
+        getStateFileContentsFromS3(
+            versionManager.getS3(),
+            versionManager.getBucketName(),
+            absoluteResourcePath,
+            stateFileName);
+    if (stateStr == null) {
+      System.out.println("No state file found in archive");
+    }
+    return stateStr;
```

Comment:
One question. So the get command won't affect the running nrtsearch, right?

Reply:
I mean I want to get the s3 state and replace the local one. Is that supported?

Reply:
These commands function out of band of the nrtsearch service. The get command will have no effect, since it is fetching the state directly from s3. The put command changes the state in the s3 backend, which will be picked up when the nrtsearch service restarts.